_includes/up-next.html

```diff
@@ -0,0 +1,162 @@
+{% assign on_current_page = false %}
+{% assign max_steps = 3 %}
+
+<section class="module up-next">
+    <article class="module-content grid-x grid-padding-x">
+        <div class="cell small-12">
+            <h6>Up Next</h6>
+        </div>
+
+
+{% for item in site.data.pages %}
+
+    {% if item.categories %}   
+
+        {% if on_current_page == true %}
+
+            {% assign next_page_ar = site.pages | where: 'permalink', item.url %}
+            {% for next_page in next_page_ar %}          
+              {% if is_next_range < max_steps && next_page.skip_index != false %}
+                
+                {%- include up-next-tile.html -%}
+
+              {% endif %}
+              {% assign is_next_range = is_next_range | plus: 1 %}
+          {% endfor %}
+        {% endif %}
+        {% if page.url == item.url %}
+            {% assign on_current_page = true %}
+            {% assign is_next_range = 0 %}
+        {% endif %}
+
+        {% for cat in item.categories %}
+
+
+                {% if cat.pages %}
+
+                        {% for pg in cat.pages %}
+
+                            {% if on_current_page == true %}
+
+                                {% assign next_page_ar = site.pages | where: 'permalink', pg.url %}
```

Comment:
There's still lots of duplicate logic throughout this file, can this be reduced further?

Reply:
I'll file a separate issue for further de-duplication.